lib/liquidoc.rb

```diff
@@ -153,16 +175,58 @@ def validate_config_structure config
 # TODO More validation needed
 end
 
+def data_hashify data_var
+  # TODO make datasource config a class
+  if data_var.is_a?(String)
+    data = {}
+    data['file'] = data_var
+    data['ext'] = File.extname(data_var)
+  else # add ext to the hash
+    data = data_var
+    data['ext'] = File.extname(data['file'])
+  end
+  return data
+end
+
+def parse_regex data_file, pattern
+  records = []
+  pattern_re = /#{pattern}/
+  @logger.debug "Using regular expression #{pattern} to parse data file."
+  groups = pattern_re.names
+  begin
+    File.open(data_file, "r") do |file_proc|
+      file_proc.each_line do |row|
+        if row.length > 3 # ignore inherently invalid lines
```

Comment:
I'm not sure this condition needs to be here, since it seems like the following `row.match(pattern_re)` call will filter out any bad lines anyway.  Doubtful, but I could see this condition eventually getting in someone's way

Reply:
I was trying to cry foul on anything that couldn't conceivably be matched by a pattern, like a line with just a `\s` or another character or two. But you're right, it's more arbitrary than leaving this open.